spec/defines/user.rb

```diff
@@ -0,0 +1,38 @@
+require 'spec_helper'
+require 'erb'
+
+describe 'prosody::user' do
+  let(:pre_condition) do
+    'include ::prosody'
+  end
+  let(:facts) do
```

Comment:
instead of mocking the facts yourself, you can you please use [rspec-puppet-facts](https://github.com/mcanevet/rspec-puppet-facts) ? you can also look at other modules, like prometheus, nginx or zabbix that allready use that pattern in their tests.